natlas-server/app/main/routes.py

```diff
@@ -48,6 +48,8 @@ def browse():
 def search():
 	''' Return search results for a given query '''
 	query = request.args.get('query', '')
+	if query == '':
+		return redirect(url_for('main.browse'))
```

Comment:
Should we provide a message that said why this happens? I'm not quite sure what this experience looks like to know if this redirect is intuitive.

Reply:
True, this would silently redirect to `/browse` which is pretty similar to what making a request to `google.com/search` does (it redirects to `google.com/webhp` for some reason lol). I could flash a message here saying that search requires a query, but the fix in the form to be `required` means that the only time this should happen now is if you typed in `<server_addr>/search` or happened to have it bookmarked from an older natlas version where `/search` had a silent default search term if one wasn't provided. The `/browse` page that it redirects to is almost identical to the `/search` except it just shows the latest results instead of showing `0 results across N hosts` that it would show if you just did a `GET /search`

I think in this case the silent redirect is fine unless you feel strongly about having a warning message.